### PR TITLE
Timestamp changes

### PR DESCRIPTION
**Pull request by sortega87**

### CODE REVIEW

**Comment by sortega87** [comment]
The following changes where made:

- Solved https://github.com/orcestra-campaign/weather/issues/15 
- The dates are now changed so that internal plots use an input the briefing time. 
- Refactor ICWV to reflect the briefing, forecast issue date, and valid date.
- Changed the first lead time to 3 hours as is the first one available in the forecasts.
- Created the module hifs.py to include common functions used to work with IFS data.

Note that the date change follows the following definitions:
- **briefing time**: The time when the briefing is generated, in UTC. This is the time we input as user through wbcli.sh
- **issued time**: The time of the last available forecast. 
- **lead hours**: The number of hours after the briefing time.
- **valid time**: The briefing time plus the lead hours.

File: src/wblib/figures/hifs.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+"""Functions to work with IFS forecasts in healpix format."""
+import pandas as pd
+
+
+FORECAST_PUBLISH_LAG = "8h"
+FORECAST_PUBLISH_FREQ = "12h"
+
+
+def get_latest_forecast_issue_time(briefing_time: pd.Timestamp):
```
Comment (HF7weatherman) [inline src/wblib/figures/hifs.py:9]: I find it a bit confusing to name the variable `briefing_time` as we provide the briefing date without a time. In my head, the briefing time is always <date>T0900 local time...
Anyways, the function does what it should, so nothing to do here.

**Review by HF7weatherman — CHANGES_REQUESTED** [review]
Thanks for doing all these changes and fixing #15. I agree with most of your changes, please see the review of the code for details.

File: src/wblib/figures/hifs.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sortega87) [reply src/wblib/figures/hifs.py:9]: I agree, and agree also that is best to leave it as it is, one way to think about it is that it is the briefing time at 00Z, and that it is a pd.Timestamp object.

File: src/wblib/figures/internal/icwv.py [hunk 2]
```diff
@@ -119,10 +106,15 @@ def _draw_icwv_current_forecast_at_lead_time(
...
-def _format_axes(current_time, lead_delta, ax):
+def _format_axes(briefing_time, issued_time, lead_delta, ax):
     lon_min, lon_max, lat_min, lat_max = FIGURE_BOUNDARIES
-    forecast_on_str = current_time.floor("1D") + lead_delta
-    ax.set_title(forecast_on_str)
+    valid_time = briefing_time + lead_delta
+    title_str = (
+        f"Forecast issue time: {issued_time.strftime('%Y-%m-%d %H:%M')} \n"
+        f"Valid time: {valid_time.strftime('%Y-%m-%d %H:%M')} \n"
+        f"Lead hours: {int(lead_delta.total_seconds() / 3600):03d}H"
```
[the remark this answers is not in the document]
Comment (sortega87) [reply src/wblib/figures/internal/icwv.py:115]: As discussed I will add the forecast issue type as an annotation to the figure. The caption will be:

"Latest IFS forecast initialization 20240801T1200UTC"

File: src/wblib/figures/internal/icwv.py [hunk 3]
```diff
@@ -133,7 +125,8 @@ def _format_axes(current_time, lead_delta, ax):
 
 
 if __name__ == "__main__":
-    time = pd.Timestamp.now("UTC").tz_localize(None)
-    lead_hours_str = "108H"
-    figure = iwv_itcz_edges(time, lead_hours_str)
-    figure.savefig("test2.png")
+    time = pd.Timestamp(2024, 8, 1).tz_localize("UTC")
```
[the remark this answers is not in the document]
Comment (sortega87) [reply src/wblib/figures/internal/icwv.py:128]: For testing, so it is better to remove it.

Note the statement `if __name__ == "__main__":` allows to import the module without executing the lines under it. Yet, should not be there.

File: src/wblib/services/get_expected_figures_yaml.py
[the remark this answers is not in the document]
Comment (sortega87) [reply src/wblib/services/get_expected_figures_yaml.py:1]: This should be now solved with the latest commit.

File: src/wblib/figures/internal/icwv.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (sortega87) [reply src/wblib/figures/internal/icwv.py]: Addressed with the latest commit

File: src/wblib/figures/internal/icwv.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (sortega87) [reply src/wblib/figures/internal/icwv.py]: Addressed with the latest commit

**Comment by sortega87** [comment]
Hi @HF7weatherman. thanks for the review and suggested changes. This should now all be addressed.

**Review by HF7weatherman — APPROVED** [review]

**Pull request merged**